### PR TITLE
adding prettier and eslint

### LINKED ISSUE #2: add prettier/eslint for typescript

### PR DESCRIPTION
**Pull request by natapokie**
## Description

**Fixes #2 **

Adding prettier and eslint for more consistent code formatting and better code practices with typesciript. For prettier, used .json file from orientation website repo (feel free to suggest other changes!)

See readme file in this pr for more info on how to configure your editor.

Make sure to also run `npm install` in the root directory to install dev dependencies needed for prettier and eslint.

## Documentation Used

- [installing eslint](https://www.npmjs.com/package/eslint)
- [installing prettier](https://prettier.io/docs/en/install#eslint-and-other-linters)
- [configure prettier with ide](https://prettier.io/docs/en/editors)
- [using eslint with prettier](https://github.com/prettier/eslint-config-prettier#installation)

## How To Test?

Once the editor is configured properly, you should see eslint warnings, e.g.,
image

On save, your code should also be autoformatted, e.g.,
`const myObj={field1:1, 'field2':2}` -> `const myObj = { field1: 1, field2: 2 };`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,21 +1,38 @@
...
 npm i
 cd ../server
 npm i
 ```
 
+This project uses [Prettier](https://prettier.io/) and [ESLint](eslint.org) for automatic formatting and fixing.
+
+Make sure to configure your [editor](https://prettier.io/docs/en/editors) to allow Prettier/ESLint on save.
+
+1. On VSCode install the Prettier and ESLint extention. Once installed and enabled, you may need to restart VSCode.
```
Comment (vicliu2001) [inline README.md:21]: there's a prettier and a prettier eslint, i want to make sure that we're supposed to get prettier and eslint seperately?
Screen Shot 2024-08-15 at 7 46 29 PM

Comment (natapokie) [reply README.md:21]: yes, two separate extensions, sorry for the confusion!
**Prettier - Code formatter** and **ESLint**

Comment (natapokie) [reply README.md]: updated the readme to make it more clear!

**Review by viiiiiicccky — APPROVED** [review]
LGTM!

**Review by vicliu2001 — APPROVED** [review]
looks good!

**Review by eriysd — APPROVED** [review]
lgtm!

**Pull request merged**